resources/components/ImageCard.vue

```diff
@@ -63,7 +70,8 @@ module.exports = {
 		return {
 			// copy suggestions from props into state so we can modify their
 			// "confirmed" properties in place
-			suggestions: this.image.suggestions
+			suggestions: this.image.suggestions,
+			hideOutline: true
```

Comment:
Will `hideOutline` ever change? If not, we should move it out of `data`.

Reply:
It changes when the user uses the tab key in this component – is there a better place to declare it `true` initially?

Reply:
I guess I don't totally understand why we can't just apply this class in a permanent/static way. But if this seems like the best way to achieve an accessibility-related UX goal then I'm fine with it.

@catrope I wonder if this kind of scenario would be a good use case for a mixin (or eventually, composition-API "hook") that keeps a small amount of internal state separate from the rest of the component, which could be included anywhere we wanted the behavior?

It seems like this is a place where "presentational state" and "logical state" are getting mixed up; ideally the ImageCard component shouldn't have to care about the `hideOutline` state even if we do need to keep track of it somehow; all this component should really care about is image prop and the state of the suggestions.

Reply:
> I guess I don't totally understand why we can't just apply this class in a permanent/static way. But if this seems like the best way to achieve an accessibility-related UX goal then I'm fine with it.
> 
The class needs to be applied initially, and stay applied, then if/when the user presses the tab key it needs to be removed and stay removed. So it's a one-way state change; the information of whether the user has ever pressed tab or not needs to be stored somewhere.

> @catrope I wonder if this kind of scenario would be a good use case for a mixin (or eventually, composition-API "hook") that keeps a small amount of internal state separate from the rest of the component, which could be included anywhere we wanted the behavior?
> 
Yes, but only if this turns out to be a common pattern. Right now I'm not super convinced that "apply/remove this CSS class when the user presses tab" is going to be a common pattern. So my take is that we shouldn't factor this out into a mixin (or composition API thingy) just yet, and wait until we see this code being duplicated enough times.

> It seems like this is a place where "presentational state" and "logical state" are getting mixed up; ideally the ImageCard component shouldn't have to care about the `hideOutline` state even if we do need to keep track of it somehow; all this component should really care about is image prop and the state of the suggestions.
Sure, but sometimes don't you just need presentational state? For example, the button component will likely need to track whether it's being pressed (a property that would be set to true on mousedown and false on mouseup) in order to apply a CSS class, that's presentational state. Is your argument that only components lower down in the component tree should have presentational state, and this component is too high-level to have presentational state?

Reply:
> Is your argument that only components lower down in the component tree should have presentational state, and this component is too high-level to have presentational state?

Less of an argument and more of a gut-feeling, but yes. This is a small codebase that probably won't have enough "levels" to separate out concerns this way, but there might be places in the future where we would want to make such distinctions (or where we'd need to re-use this kind of behavior).